### ISSUE TITLE
Cannot extend specific environments

### ISSUE THREAD

**Issue by steverep** [browserslist/browserslist#832]
I find the feature to extend configs pretty limiting.  Not only can you not extend other local files (#408), but there is no way to extend specific environments from the shared config or from the same config.  Extending from another environment in the same config could be especially useful.  For example:

```
[legacy]
supports es5 and not dead

[legacyPlus]
extends legacy from . and supports clipboard
```

I'll note that a shared config can have different environments that can be selected, but the method for doing so is confusing and doesn't work unless the environment name in the local config matches that of the shared config.  For example, borrowing the syntax above, you cannot do:

```
[legacyPlus]
extends legacy from sharedConfig and supports clipboard
```

I think there are several possible syntax changes that could easily add this:
- "extends _env_ from _config_", where _config_ can be "." or another keyword to represent the current config
- "extends _value_", where _value_ could also be the name of another environment in the same config in addition to the current behavior
- Add a new "_env_ environment" query

My gut says the first one is best but I'd need to examine the code further.

**Comment by ai** [comment]
Honestly, it looks like a very complex feature for very rare use case.

Browserslist configs is just a few lines of code, you can do

```
[legacy]
supports es5 and not dead

[legacyPlus]
supports es5 and not dead and supports clipboard
```

**Comment by steverep** [comment]
I agree it's probably on the niche side of use cases, but would be easier to work with and comprehend complex cases.  Repetition is always prone to become a footgun of course.

Mostly though, I just see the use of an `env` option and an `extends` together as confusing.  I had to dig into the code to see that it just passes the same `env` used in the local config directly to the shared config.  So if they aren't the same and you don't have `throwOnMissing = true`, then the results might be unexpected.
